### PR TITLE
Migrate from Truffle to Hardhat

### PR DESCRIPTION
**Pull request by Arachnid**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,27 +1,31 @@
...
   "requires": true,
   "lockfileVersion": 1,
   "devDependencies": {
     "@ensdomains/buffer": "^0.0.8",
     "@ensdomains/solsha1": "^0.0.2",
+    "@nomiclabs/hardhat-truffle5": "^2.0.0",
+    "@nomiclabs/hardhat-web3": "^2.0.0",
+    "@openzeppelin/test-helpers": "^0.5.10",
     "@truffle/hdwallet-provider": "^1.2.1",
     "dnsprovejs": "0.0.9",
```
Comment (makoto) [inline package.json:13]: The latest dnsprovejs is 0.3.7

Comment (makoto) [reply package.json:13]: Actually, it's `@ensdomains/dnsprovejs`.  You are more likely importing wrong package

File: truffle.js [hunk 2]
```diff
@@ -1,41 +0,0 @@
...
-      gasPrice: 1
-    }
-  },
-  compilers: {
-    solc: {
-      version: "0.7.4",
-    }
-  },
-  plugins: [
-    'truffle-plugin-verify'
```
Comment (makoto) [inline truffle.js:36]: Does hardhat handle etherscan verification automatically? Looks like hardhat has its own `verify` task which you may want to put into package.json https://hardhat.org/plugins/nomiclabs-hardhat-etherscan.html#tasks

File: truffle.js [hunk 3]
```diff
@@ -1,41 +0,0 @@
...
-      },
-      // host: 'localhost',
-      // port: 8545,
-      network_id: '3',
-      from: '0xa303ddc620aa7d1390baccc8a495508b183fab59'
-      // gas: 5000000
-    }
-  },
-  mocha: {
-    reporter: 'eth-gas-reporter',
```
Comment (makoto) [inline truffle.js:24]: Same for this. Does hardhat offer similar service? If not, you may want to get rid of `eth-gas-reporter` package

File: hardhat.config.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+require("@nomiclabs/hardhat-truffle5");
+
+module.exports = {
+  networks: {
```
Comment (makoto) [inline hardhat.config.js:4]: How do we deploy now? You don't need to put public testnet specific info which you had on `truffle.js` for Ropsten

**Comment by makoto** [comment]
@Arachnid  can you update README? It still mentions `truffle compile` and `truffle test` .

**Comment by makoto** [comment]
Also mentions that it now uses hardhat

File: truffle.js (hunk 2 above)
Comment (Arachnid) [reply truffle.js:36]: I'll fix this later for Hardhat once we're ready to deploy.

File: hardhat.config.js (hunk 4 above)
Comment (Arachnid) [reply hardhat.config.js:4]: I'll need to write a new deployment script when the time comes.

**Pull request merged**